pontoon/contributors/static/js/settings.js

```diff
@@ -13,27 +18,23 @@ $(function () {
 
     const self = $(this);
     const value = self.val().trim();
-    const attribute = self.data('attribute');
-    const originalValue = self.data('original-value') || '';
+    const attribute = this.dataset.attribute;
+    const originalValue = this.dataset.originalValue || '';
 
     if (value === originalValue) {
       return;
     }
 
+    this.dataset.originalValue = value;
```

Comment:
Isn't this presuming success? If you get an error response from the POST, shouldn't the original value be retained?